client/my-sites/checkout/cart/cart-item.jsx

```diff
@@ -65,30 +68,49 @@ export class CartItem extends React.Component {
 
 	monthlyPrice() {
 		const { cartItem, translate } = this.props;
-		const { cost, currency } = cartItem;
+		const { currency } = cartItem;
 
-		if ( typeof cost === 'undefined' ) {
+		if ( ! this.monthlyPriceApplies() ) {
 			return null;
 		}
 
+		const { months, monthlyPrice } = this.calcMonthlyBillingDetails();
+
+		return translate( '(%(monthlyPrice)s %(currency)s x %(months)d months)', {
+			args: {
+				months,
+				currency,
+				monthlyPrice: monthlyPrice.toFixed( currency === 'JPY' ? 0 : 2 ),
+			},
+		} );
+	}
+
+	monthlyPriceApplies() {
+		const { cartItem } = this.props;
+		const { cost } = cartItem;
+
 		if ( ! isPlan( cartItem ) ) {
-			return null;
+			return false;
 		}
 
-		if ( cost <= 0 ) {
-			return null;
+		if ( isMonthly( cartItem ) ) {
+			return false;
 		}
 
-		if ( isMonthly( cartItem ) ) {
-			return null;
+		const hasValidPrice = typeof cost !== 'undefined' && cost > 0;
```

Comment:
It would be more readable if `typeof cost !== 'undefined'` is replaced with `! isNaN(cost)`. 
[`isNaN`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN) returns `true` when `undefined` is passed.